Change references from RFC 5245 to RFC 8445 and I-D.ietf-mmusic-ice-sip-sdp.

pthatcherg opened the pull request:
Fixes #841 
Fixes #852

juberti commented on draft-ietf-rtcweb-jsep.xml:
```diff
@@ -1932,6 +1934,10 @@ candidate:1 1 UDP 1694498815 192.0.2.33 10000 typ host
             be added, as specified in
             <xref target="I-D.ietf-ice-trickle"></xref>, Section 4.</t>
 
+            <t>An "a=ice-options" line with the "ice2" option MUST
```
ice2 should be combined into the previous paragraph that discusses ice-options.

pthatcherg replied:
Whoops, duh. Fixed.

juberti commented on draft-ietf-rtcweb-jsep.xml:
```diff
@@ -2391,7 +2400,7 @@ candidate:1 1 UDP 1694498815 192.0.2.33 10000 typ host
             <xref target="sec.ice-gather-overview"></xref>), an
             "a=candidate" line MUST be added, as defined in
-            <xref target="RFC5245"></xref>, Section 4.3., paragraph 3.
+            <xref target="8445"></xref>, Section 4.1.
```
This reference doesn't seem right - there is no S 4.1 in 8445.

juberti replied:
Seems like this should be a ref to sip-sdp

pthatcherg replied:
Yep; fixed

juberti commented on draft-ietf-rtcweb-jsep.xml:
```diff
@@ -2595,7 +2605,9 @@ candidate:1 1 UDP 1694498815 192.0.2.33 10000 typ host
-          <xref target="I-D.ietf-ice-trickle"></xref>, Section 4, is
+          <xref target="I-D.ietf-ice-trickle"></xref>, Section 4,
+          and the "ice2" option as specified in
+          <xref target="RFC8445"></xref>, Section 10, is
```
is->are

pthatcherg replied:
I think the subject is 'the "a=ice-options" line', which is singular.

juberti commented on draft-ietf-rtcweb-jsep.xml:
```diff
@@ -3751,13 +3767,14 @@ candidate:1 1 UDP 1694498815 192.0.2.33 10000 typ host
           <t>If the ICE ufrag or password changed from the previous
           remote description:
-          <xref target="RFC5245" />.
+          <xref target="RFC8445" />.
```
This line is a typo and should just be removed.

pthatcherg replied:
Done

juberti commented on draft-ietf-rtcweb-jsep.xml:
```diff
@@ -3567,7 +3584,7 @@ candidate:1 1 UDP 1694498815 192.0.2.33 10000 typ host
               <t>ICE ufrag and password values, which MUST comply with
               the size limits specified in
-              <xref target="RFC5245" />, Section 15.4.</t>
+              <xref target="RFC8445" />, Section 5.3.</t>
```
These limits don't seem clear enough - is the max length specified somewhere?

pthatcherg replied:
Ah, it's in ietf-mmusic-ice-sip-sdp 4.4.  I've fixed the ref.

pthatcherg commented:
OK, I think I fixed everything

juberti commented:
@fluffy, that makes sense. That probably needs to be new text in Section 5.11, I think that could be separate from this PR (which doesn't talk about which ICE version is used at all).

fluffy commented:
I don't care if it is one PR to two and the rest of this PR looks good to me.

pthatcherg commented:
I added a note about RFC 5245 (based on Adam's email) and I added text like what Cullen was asking for around accepting offers and answers without "ice2".  I excluded the part of Adam's not about previous versions of trickle ICE because it didn't seem relevant to this PR and it had a missing ref and I didn't know what the ref should be (so I remove it).

TL;DR: I think this is closer to what we want eventually.  I'd suggest we land something and then tweak if further in a followup PR.  But I'm willing to delete or modify the note if that would get the PR landed.

adamroach commented on draft-ietf-rtcweb-jsep.xml:
```diff
@@ -148,7 +148,17 @@ ipr="trust200902">
+        While this specification formally relies on <xref
+        target="RFC8445"></xref>, at the time of its publication, the
+        majority of WebRTC implementations support the version of ICE
+        described in <xref target="RFC5245"></xref>, the use of the
```
This is a comma splice.
```suggestion
        described in <xref target="RFC5245"></xref>. The use of the
```

juberti commented on draft-ietf-rtcweb-jsep.xml:
```diff
@@ -3639,18 +3664,17 @@ candidate:1 1 UDP 1694498815 192.0.2.33 10000 typ host
-          <xref target="RFC5245" />, Section 9.1.1.1. If this
+          trigger the ICE agent to start an ICE restart as described in
+          <xref target="RFC8445" />, Section 9, and begin
+          gathering new candidates for the m= section . If this
```
extra space
